src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs

```diff
@@ -25,9 +25,12 @@ public static partial class DataContractJsonSerializerTests
 
     static DataContractJsonSerializerTests()
     {
-        var method = typeof(DataContractSerializer).GetMethod(SerializationOptionSetterName, BindingFlags.NonPublic | BindingFlags.Static);
-        Assert.True(method != null, $"No method named {SerializationOptionSetterName}");
-        method.Invoke(null, new object[] { 1 });
+        if (!PlatformDetection.IsFullFramework)
+        {
+            MethodInfo method = typeof(DataContractSerializer).GetMethod(SerializationOptionSetterName, BindingFlags.NonPublic | BindingFlags.Static);
+            Assert.True(method != null, $"No method named {SerializationOptionSetterName}");
+            method.Invoke(null, new object[] { 1 }); 
+        }
```

Comment:
For NetCore, we use ILGen to generate serializer assembly. For UWP, ILGen is not available at runtime, so we use the UWP toolchain to pre-generate serializer assemblies. One of the limitations with pre-generation is that it cannot support any features required runtime values, e.g. [`KnownTypeAttribute.MethodName`](https://msdn.microsoft.com/en-us/library/system.runtime.serialization.knowntypeattribute.methodname(v=vs.110).aspx) is not supported on UWP. To overcome this limitation, we enable the serializer to serialize/de-serialize using reflection at runtime (we call it refleciton-based serialization). The test project under `corefx/src/System.Runtime.Serialization.Json/tests/ReflectionOnly` is for covering the new serialization method.

The reflection-based serialization is not available on Desktop, so we are exclude it from running on full framework.